web3c/confidential.js

```diff
@@ -10,89 +12,107 @@ const Signer = require('./signer');
  * and exposes the confidential contract interface by creating a
  * web3.eth.Contract communication through a ConfidentialProvider.
  */
-const Confidential = function (web3, storage, mraebox) {
-  Object.assign(this, web3.eth);
-  this.__proto__ = web3.eth.__proto__;
-
-  this.keyManager = new KeyManager(web3, storage, mraebox);
-  let provider = new ConfidentialProvider(this.keyManager, web3._requestManager);
-  Confidential.methods(web3.extend).forEach((method) => {
-    method.setRequestManager(web3._requestManager);
-    method.attachToObject(this);
-  });
-
-  const defaultTransformer = new ConfidentialProvider.private.ConfidentialSendTransform(provider, this.keyManager);
-  wrapAccounts(this.accounts, defaultTransformer);
-
-  // Save `this` so that we can refer to it and its properties inside `ConfidentialContract`.
-  // Otherwise `this` is overridden when `new` is used in `new Contract`.
-  let self = this;
+class Confidential {
+  constructor (web3, storage, mraebox) {
+    this.keyManager = new KeyManager(web3, storage, mraebox);
+    this.setupConfidentialRpc(web3);
+    this.setupConfidentialContract(web3, storage, mraebox);
+    this.setupWrappedAccounts(web3);
+  }
+
   /**
-   * web3.confidential.Contract behaves like web3.eth.Contract.
-   * @param {Object} abi
-   * @param {String} address
-   * @param {Object} options
-   * @param {String} options.key The longterm key of the contract.
-   * @param {bool}   options.saveSession false to disable storing keys.
+   * Creates methods on the confidential namespace that make requests to the confidential_*
+   * web3c rpc endpoints. For example, one may do `web3c.confidential.getPublicKey(address)`.
    */
-  this.Contract = function ConfidentialContract(abi, address, options) {
-    let c = new web3.eth.Contract(abi, address, options);
-    // Copy the wrapped contract to `this`.
-    Object.assign(this, c);
-    this.__proto__ = c.__proto__;
-
-    // Object.DefineProperty's are not copied otherwise.
-    this.defaultAccount = c.constructor.defaultAccount;
-    this.defaultBlock = c.constructor.defaultBlock || 'latest';
-
-    let instanceProvider = provider;
-
-    let keymanager = self.keyManager;
-    if (options && options.saveSession === false) {
-      keymanager = new KeyManager(web3, undefined, mraebox);
-      instanceProvider = new ConfidentialProvider(keymanager, web3._requestManager);
-    }
-
-    c.setProvider.call(this, instanceProvider);
-
-    // Deployed contracts are instantiated with clone.
-    // This patch keeps those clones confidential.
-    this.clone = () => {
-      return new ConfidentialContract(this.options.jsonInterface, this.options.address, this.options);
-    };
+  setupConfidentialRpc(web3) {
```

Comment:
the methods as a static method on confidential was setup to parallel the structure used by web3.eth.

I don't think we care about that, especially given the restructuring of that library, but there was reason for why it was structured in a somewhat convoluted way :)

Reply:
We shouldn’t copy the subjectively annoying syntax of another code base just for the sake of it :P

This way it’s much easier to understand what’s going on  for people not used to js

Reply:
Changed all these methods to `_setup*`.